atest/testdata/standard_libraries/builtin/wait_until_keyword_succeeds.robot

```diff
@@ -115,3 +125,13 @@ Wait Until Inside User Keyword
 Timeouted UK with Wait Until KW
     [Timeout]    ${timeout}
     Wait Until Keyword Succeeds    100ms    10ms    Fail    Error in timeouted UK
+
+Get Length On Returned Variable Works On Last Attempt
+    ${myvar} =    return_value_after_retries    Hello
+    Get Length    ${myvar}
+    Fail    Test PASSES : variable is no more None
```

Comment:
I looked at the existing tests in this suite and was surprised that they didn't verify anything on the running side. I decided to add that for few tests because I think that's a good idea in general and because they can act as examples for you.

Tests were enhanced in 12b646aeec291565ce6b1378223722fd2356f0f6. I hope that doesn't cause merge conflicts.